staging/src/k8s.io/client-go/pkg/api/types.go

```diff
@@ -1149,6 +1155,36 @@ type ScaleIOVolumeSource struct {
 	ReadOnly bool
 }
 
+// Represents a StorageOS persistent volume resource.
+type StorageOSVolumeSource struct {
+	// VolumeName is the human-readable name of the StorageOS volume.  Volume
+	// names are only unique within a namespace.
+	VolumeName string
+	// VolumeNamespace specifies the scope of the volume within StorageOS.  If no
+	// namespace is specified then the Pod's namespace will be used.  This allows the
+	// Kubernetes name scoping to be mirrored within StorageOS for tighter integration.
+	// Set VolumeName to any name to override the default behaviour.
+	// Set to "default" if you are not using namespaces within StorageOS.
+	// Namespaces that do not pre-exist within StorageOS will be created.
+	// +optional
+	VolumeNamespace string
+	// The name of the storage pool to provision from.  Pools can have different
+	// capacity and performance characteristics depending on the underlying
+	// storage resources used in them.  If not specified, the default pool will be
+	// used.
+	// +optional
+	Pool string
```

Comment:
I took a quick look at the current plugins:
* gluster doesn't have storage pool in its API
* rbd does have pool in its API, but it needs it for mounting
* scaleio does have it and it seems like its only used for provisioning, so we missed that one

Saad's main point is that if the storage pool is only used for provisioning, and not used for mounting or attaching, then it should just be a parameter in the StorageClass and not in the volume source API.